tests/functional/object/rgw/test_rgw_sts.py

```
@@ -524,3 +520,306 @@ def test_rgw_sts_configuration(
+        logger.info("Step 5: Listing bucket content using invalid Access Key")
+        invalid_access_key_client = self.make_boto3_client(
+            "s3",
+            endpoint,
+            region,
+            "INVALIDACCESSKEY00000",
+            assumed_secret_key,
+            assumed_session_token,
+        )
+        try:
+            invalid_access_key_client.list_objects_v2(Bucket=bucket_name)
+            assert False, "Expected error with invalid Access Key but got success"
```

Ruff (0.15.15)

[warning] Assertion always fails, replace with `pytest.fail()`

(PT015)